hooks/pre_validations/05-plugins.rb

```diff
@@ -0,0 +1,87 @@
+if app_value(:list_plugins)
+
+  class Plugin
+    include Comparable
+
+    attr_reader :group, :plugin, :provider, :module
+
+    # We expect a mymod::plugin::myplugin structure but also allow
+    # mymod::plugin::myplugin::provider
+    def initialize(mod)
+      parts = mod.identifier.split('::')
+
+      raise ArgumentError, "#{mod.identifier} is not a plugin" unless parts.length >= 3
+
+      type = ['plugin', 'cli', 'compute'].detect { |type| parts.include?(type) }
+      raise ArgumentError, "#{mod.identifier} is not a plugin" unless type
+
+      index = parts.index(type)
+      if type == 'plugin'
+        @group = parts.slice(0, index).join(' ')
+      else
+        @group = parts.slice(0, index + 1).join(' ')
+      end
+      @plugin, @provider = parts.slice(index + 1, parts.length)
+
+      @module = mod
+    end
+
+    def name
+      provider || plugin
+    end
+
+    def to_s
+      mod.identifier
+    end
+
+    def inspect
+      "<Plugin:#{mod.identifier}>"
+    end
```

Comment:
There aren't actually used and I couldn't quite get them to work in debugging. Perhaps they should be removed.